python/tank/context.py

```diff
@@ -233,6 +248,7 @@ def __deepcopy__(self, memo):
         ctx_copy.__task = copy.deepcopy(self.__task, memo)
         ctx_copy.__user = copy.deepcopy(self.__user, memo)        
         ctx_copy.__additional_entities = copy.deepcopy(self.__additional_entities, memo)
+        ctx_copy.__source_entity = copy.deepcopy(self.__source_entity)
```

Comment:
Please pass in memo so next person who adds something there doesn't get caught in a bug.